nsq/message.py

```diff
@@ -51,13 +51,22 @@ class Message(event.EventedMixin):
     def __init__(self, id, body, timestamp, attempts):
         self._async_enabled = False
         self._has_responded = False
+
         self.id = id
         self.body = body
         self.timestamp = timestamp
         self.attempts = attempts
+        self.timed_out = False
 
         super(Message, self).__init__()
 
+    def is_alive(self):
```

Comment:
This method offers a second approach to users for acting on the fact their message has timed out. If, for example, a message is timed out and about to undergo an expensive operation; the user can check `is_alive` and simply discard the message. 

The alternative using callbacks would require some significant overhead: either a registry of messages that have timed out so the same thing can be accomplished, or setting a similar flag in the callback itself. Imho neither of those two are ideal, so offering this interface gives the user a simple alternative.